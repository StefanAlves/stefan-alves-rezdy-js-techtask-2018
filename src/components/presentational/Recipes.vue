<!-- src/components/Recipes.vue -->
<template>
  <div>
    <recipe v-for="(recipe, index) in recipes" :key="index" :recipe="recipe" />  
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from "vue-property-decorator";
import Recipe from "./Recipe.vue";

@Component({
  components: { Recipe }
})
export default class Recipes extends Vue {
  @Prop()
  recipes!: string;
}
</script>