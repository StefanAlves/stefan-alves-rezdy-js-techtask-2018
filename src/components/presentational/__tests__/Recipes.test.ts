import { shallowMount, mount, config } from '@vue/test-utils';
import Recipes from '../Recipes.vue';

config.logModifiedComponents = false

describe('Recipes', () => {
  const recipes = [
    {
      title: 'Fry-up',
      ingredients: [
          'Bacon',
          'Eggs'
      ]
    },
    {
      title: 'Salad',
      ingredients: [
        'Lettuce',
        'Tomato'
      ]
    }
  ];
  

  it('renders correctly', () => {
    const wrapper = mount(Recipes, {
        propsData: { recipes }
    });
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render', () => {
    const wrapper = shallowMount(Recipes, {
        propsData: { recipes }
    })
    const recipeStubs = wrapper.findAll('recipe-stub');
    expect(recipeStubs.length).toEqual(2)
  });
});