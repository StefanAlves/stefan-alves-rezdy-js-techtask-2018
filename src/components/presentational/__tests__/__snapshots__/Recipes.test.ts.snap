// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes renders correctly 1`] = `
<div>
  <div
    class="recipe"
  >
    <div
      class="title"
    >
      Fry-up
    </div>
     
    <ul
      class="ingredients"
    >
      <li
        class="ingredient"
      >
        
      Bacon
    
      </li>
      <li
        class="ingredient"
      >
        
      Eggs
    
      </li>
    </ul>
  </div>
  <div
    class="recipe"
  >
    <div
      class="title"
    >
      Salad
    </div>
     
    <ul
      class="ingredients"
    >
      <li
        class="ingredient"
      >
        
      Lettuce
    
      </li>
      <li
        class="ingredient"
      >
        
      Tomato
    
      </li>
    </ul>
  </div>
</div>
`;
